### PR TITLE
Added wiki style user docs

### PR DESCRIPTION
**Pull request by basshelal**
Still a work in progress but creating this to keep communication and for logging.

This is in relation to #42 

The general idea is to make the project look more appealing and approachable to beginners and more professional and polished.

JNA does this much better than us and we should at least *try* to approach their level of documentation.

I'll try to add as much as I possibly can with the knowledge I have, hopefully I (and others) add to it in the future with more experience using the library.

@headius Let me know if I'm completely off base with this idea or if I should do anything completely differently before I go further down the rabbit-hole

### CODE REVIEW

**Comment by basshelal** [comment]
Oh I also intend/want to update and improve the JavaDocs of the source code itself but that will take a lot more time and a much deeper knowledge of the inner workings of the code so I'll tackle that after this in another PR.

**Comment by headius** [comment]
Wow this is great! I will review and merge.

**Comment by headius** [comment]
Oh just noticed it is draft. Let me know when you are ready to merge!

**Comment by basshelal** [comment]
@headius 
Ok I think this is a good starting point, it covers the basics more or less.

I'd want to improve and add more when I'm more confident with the intricate details of the codebase, for example I thought you can use `LibraryOption.LoadNow` to load the library immediately rather than lazily but looking through the code `LibraryOption.LoadNow` is unused, this is mentioned in #130. I didn't look through to see what the actual behavior is so I deleted any reference to that until I understand more and am more confident.

Let me know if there's anything I was wrong about and of course feel free to change it and add to it yourself. I just kind of wanted to make this library more polished looking and professional because my experience with it has been phenomenal but it's difficult not to be annoyed at the lack of documentation and guidance there is, especially for something so well made, it kind of deters anyone with less experience or patience.

Anyway, hope this helps 😊

**Comment by basshelal** [comment]
@headius By the way the PR is ready to merge in case I didn't make it clear earlier

**Comment by headius** [comment]
I will look over and merge!

**Review by headius — APPROVED** [review]
Great content here, though I wonder how we might unify in-repo docs with the wiki. 🤔

**Comment by headius** [comment]
Looks great, thank you! I do wonder if there's a good way we could have this live in the wiki **and** be in the repo, but that is more a GitHub question.

**Comment by basshelal** [comment]
@headius Yeah I thought about the same thing! To be honest I don't think it's that necessary. I've never really liked GitHub's wikis at all, they add nothing over just using regular markdown files and are too GitHub focused. With regular markdown, you can view offline, easily commit PRs and integrates with git etc *and* you still get the nice rendering of a Markdown page on GitHub. I mean even JNA don't use a wiki and just have a directory of markdown files like I've done (I definitely did not copy that idea from them at all! 😶)

Unless it's a super easy, do it once never think about it again thing, I wouldn't bother. Just a quick Google search shows me some Github Actions that kind of do something like syncing but that's just a hassle.

The best thing you can do right now, is just make the wiki link back to the `docs` directory. I'll see if I can do that myself now.

Did you read the contents? Is everything I wrote correct? I wanted to add more but as you can see from my commits I ended up deleting things that I thought were right but turned out to be wrong or questionable at best. When I have understood more of the internals and feel more confident I'll try and add more to the docs.

Thanks for merging, really appreciate your welcoming attitude and I'm really happy to have helped again 😊!

**Comment by headius** [comment]
@basshelal I did read through it quickly and everything looks pretty good. I may do some cleanup of a few sections and try to clarify further when we think JNR is a better option than JNA or Panama, but it is a great start!

I will also try to have folks like @enebo and @kares review it since they work with the JNR libraries in JRuby quite often.

File: docs/SqueezingPerformance.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Squeezing Performance
+
+Performance should already be good enough for 99% of people, but to squeeze out more when necessary, there are some
+tricks.
+
+These are ordered from least to most inconvenient (and drastic)
+
+## `@IgnoreError`
+
+A huge performance improvement can be gained by telling JNR-FFI to note save the last errno. By default, JNR-FFI will
```
Comment (enebo) [inline docs/SqueezingPerformance.md:10]: note should be not

**Review by enebo — COMMENTED** [review]
Looks really really good.  I saw one typo so fix that but otherwise I say "ship it"!

File: docs/SqueezingPerformance.md (hunk 1 above)
Comment (headius) [reply docs/SqueezingPerformance.md:10]: I already merged but will fix that typo in place.

**Pull request merged**